src/cq.rs

```diff
@@ -43,6 +85,22 @@ impl Drop for CompletionQueueHandle {
     }
 }
 
+pub struct CompletionQueueRef<'a> {
+    queue: &'a CompletionQueue,
+}
+
+impl<'a> CompletionQueueRef<'a> {
+    pub fn as_ptr(&self) -> *mut GrpcCompletionQueue {
```

Comment:
The `ptr` may still outlive `CompletionQueueRef`.

Reply:
Yes. Because there is no way to trace the lifetime of a raw pointer, so I guess that's the best guarantee we can get. Just like `Vec::as_ptr` can't provide any promise of lifetime of the returned value. `CompletionQueueRef` here just ensures the completion queue will not be shutdown when ref is still valid.